### PR TITLE
[SPARK-10913][SPARKR] attach() function support

### PR DESCRIPTION
**Pull request by adrian555**

### CODE REVIEW

**Comment by shivaram** [comment]
Jenkins, add to whitelist

**Comment by shivaram** [comment]
Jenkins, ok to test

**Comment by adrian555** [comment]
Could admin please take a look at the error caught by jenkins run? It does not seem to relate with any change from this PR.

**Comment by shivaram** [comment]
Jenkins, retest this please

**Comment by shivaram** [comment]
@adrian555 Could you bring this branch up to date with master ? The PR doesn't merge correctly right now

File: R/pkg/inst/tests/test_sparkSQL.R [hunk 1]
```diff
@@ -1327,6 +1327,15 @@ test_that("SQL error message is returned from JVM", {
   expect_equal(grepl("Table Not Found: blah", retError), TRUE)
 })
 
+test_that("attach() on a DataFrame", {
+  df <- jsonFile(sqlContext, jsonPath)
+  expect_error(age)
+  attach(df)
+  expect_output(age, "DataFrame[age:bigint]", fixed=T)
```
Comment (felixcheung) [inline R/pkg/inst/tests/test_sparkSQL.R]: could you please add more test on how it could be used?

**Comment by felixcheung** [comment]
if we add attach shouldn't we add detach?
https://stat.ethz.ch/R-manual/R-devel/library/base/html/attach.html

Also, isn't `with` better for this so as to not leak the reference

**Comment by adrian555** [comment]
I created a new pull request #8997 to address Shivaram's comment.

@felixcheung the R native detach() works so we do not need SparkR version of detach(). The with() is another function I am planning to work on after this one. I think we need both.

**Pull request closed without merging**